datanode/broker/buffered_event_source.go

```diff
@@ -322,8 +380,10 @@ func readEvent(eventFile *os.File, offset int64) (event *eventspb.BusEvent, seqN
 	return event, seqNum, totalBytesRead, nil
 }
 
+const bufferFileNamePrepend = "datanode-buffer"
+
 func (m *FileBufferedEventSource) getBufferFileName(fromSeqNum uint64, toSeqNum uint64) string {
-	return fmt.Sprintf("%s/datanode-buffer-%d-%d", m.bufferFilePath, fromSeqNum, toSeqNum)
+	return fmt.Sprintf("%s/%s-%d-%d.bevt", m.bufferFilePath, bufferFileNamePrepend, fromSeqNum, toSeqNum)
```

Comment:
I think you need to specify the number of digits for the integer values here, otherwise the sorting of the filenames can be corrupted: e.g.

prefix-1.ext
prefix-10.ext
prefix-2.ext

etc.

Reply:
I prepended the time to ensure the sort works, the seq num is really just for info, it serves no purpose

Reply:
that is to say, the time is preppended on the archive files, you don't need it on the non archived files as the buffer reader knows what file nam,e to expect